### PR TITLE
Prepare removal of feature flag 'dynamic-heap-size'

### PR DESCRIPTION
**Pull request by bjorncs**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java [hunk 1]
```diff
@@ -311,7 +309,6 @@ public FeatureFlags(FlagSource source, ApplicationId appId, Version version) {
         }
         @Override public boolean alwaysMarkPhraseExpensive() { return alwaysMarkPhraseExpensive; }
         @Override public int contentLayerMetadataFeatureLevel() { return contentLayerMetadataFeatureLevel; }
-        @Override public boolean dynamicHeapSize() { return dynamicHeapSize; }
```
Comment (baldersheim) [inline configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java:314]: Keep this and the the flag with updated default.
Then GC flag override this version is rolled out.
Then you can remove this and the flag definition afterwards

File: flags/src/main/java/com/yahoo/vespa/flags/Flags.java [hunk 2]
```diff
@@ -385,8 +385,8 @@ public class Flags {
             INSTANCE_ID);
 
     public static final UnboundBooleanFlag DYNAMIC_HEAP_SIZE = defineFeatureFlag(
-            "dynamic-heap-size", false,
-            List.of("bjorncs"), "2023-09-21", "2024-02-15",
+            "dynamic-heap-size", true,
+            List.of("bjorncs"), "2023-09-21", "2024-01-15",
```
Comment (baldersheim) [inline flags/src/main/java/com/yahoo/vespa/flags/Flags.java:389]: I think you need to keep the future date of February 1st

File: config-model-api/src/main/java/com/yahoo/config/model/api/ModelContext.java [hunk 3]
```diff
@@ -109,7 +109,7 @@ interface FeatureFlags {
         @ModelFeatureFlag(owners = {"arnej"}) default String logFileCompressionAlgorithm(String defVal) { return defVal; }
         @ModelFeatureFlag(owners = {"baldersheim"}, comment = "Select summary decode type") default String summaryDecodePolicy() { return "eager"; }
         @ModelFeatureFlag(owners = {"vekterli"}) default int contentLayerMetadataFeatureLevel() { return 0; }
-        @ModelFeatureFlag(owners = {"bjorncs"}) default boolean dynamicHeapSize() { return false; }
+        @ModelFeatureFlag(owners = {"bjorncs"}, removeAfter = "8.288") default boolean dynamicHeapSize() { return true; }
```
Comment (baldersheim) [inline config-model-api/src/main/java/com/yahoo/config/model/api/ModelContext.java:112]: Given the below comment, you probably need 289 here.

**Review by baldersheim — DISMISSED** [review]

File: configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java (hunk 1 above)
Comment (bjorncs) [reply configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java:314]: The interface's default implementation defines the new default value. There is no need for keeping the wiring anymore.

Comment (baldersheim) [reply configserver/src/main/java/com/yahoo/vespa/config/server/deploy/ModelContextImpl.java:314]: Yes, that is correct. But you now leave a flagdefinition file that has no effect, other that than preventing an error messeage. That is why I like to keep the wiring one version longer and remove flag def and wiring in same PR.

**Review by hmusum — APPROVED** [review]

**Pull request merged**